### PR TITLE
Incorrect notification image after fast switch

### PR DESCRIPTION
**Pull request by mvn-hungtruong-dn**
https://developer.android.com/develop/ui/views/notifications#limits
Android limits the notification update rate to 5, so if users tap the buttons recursively the notification won't be updated
This is the cause of the issue
https://github.com/doublesymmetry/react-native-track-player/issues/2021

So I have to limit the debounce of the invalidate call and also invalidate once again after 1s to make sure the UI is up-to-date.
I also add the parameter `androidNotificationDebounceInterval` for `react-native-track-player` to use.

Please review `NotificationManager` and `BaseAudioPlayer`. If it is OK. I will create another PR from this repository instead of my forked repo. Thanks. 😄

### CODE REVIEW

**Comment by kirillzyusko** [comment]
@puckey we are using 300ms as a debounce interval.

**Comment by puckey** [comment]
Why is the extra update after 1 second necessary? Which case does it solve?

**Comment by mvn-hungtruong-dn** [comment]
@puckey Because even with 300ms, if the user switches fast enough, there will be a chance to reproduce this issue.
So 1 second after the user stops switching, we will invalidate again to make sure everything in the notification is up-to-date.

File: kotlin-audio/build.gradle [hunk 1]
```diff
@@ -5,17 +5,17 @@ plugins {
     id 'de.mannodermaus.android-junit5'
 }
 
-def versionNumber = '2.0.0'
+def versionNumber = '1.0.0'
```
Comment (jspizziri) [inline kotlin-audio/build.gradle:8]: Why are you rolling back the version number?

File: .idea/.gitignore [hunk 2]
```diff
@@ -1,3 +0,0 @@
-# Default ignored files
-/shelf/
```
Comment (jspizziri) [inline .idea/.gitignore:2]: Why are you deleting these files? Doesn't seem relevant to the fix.

File: kotlin-audio/build.gradle [hunk 3]
```diff
@@ -5,17 +5,17 @@ plugins {
...
 
 group = 'com.github.doublesymmetry'
 version = versionNumber
 
 android {
-    compileSdk 31
+    compileSdk 33
```
Comment (jspizziri) [inline kotlin-audio/build.gradle:14]: Bumping these versions create a BC break. Is this absolutely necessary? I'm not crazy about it.

File: kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/models/PlayerConfig.kt [hunk 4]
```diff
@@ -1,7 +1,6 @@
 package com.doublesymmetry.kotlinaudio.models
 
-import com.google.android.exoplayer2.C
```
Comment (jspizziri) [inline kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/models/PlayerConfig.kt:3]: Were these unused imports?

File: kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/notification/NotificationManager.kt [hunk 5]
```diff
@@ -321,21 +352,26 @@ class NotificationManager internal constructor(
...
+
                     is NotificationButton.FORWARD -> {
                         forwardIcon = button.icon ?: forwardIcon
                         PlaybackStateCompat.ACTION_FAST_FORWARD
                     }
+
                     is NotificationButton.SEEK_TO -> {
                         PlaybackStateCompat.ACTION_SEEK_TO
                     }
+
```
Comment (jspizziri) [inline kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/notification/NotificationManager.kt:369]: Please don't intermingle stylistic changes with bug fixes. Can you toll all these style changes back?

**Review by jspizziri — CHANGES_REQUESTED** [review]

**Comment by mvn-hungtruong-dn** [comment]
@jspizziri I already notified this in the description. Please review `NotificationManager` and `BaseAudioPlayer`, if it's OK I will create another PR from this repository instead of my forked repository.
I will also revert the style in both files. 😄

File: kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/models/PlayerConfig.kt (hunk 4 above)
Comment (mvn-hungtruong-dn) [reply kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/models/PlayerConfig.kt:3]: Yes, it is.

**Comment by mvn-hungtruong-dn** [comment]
Please review this PR instead.

**Pull request closed without merging**